javascript.md

```diff
@@ -75,6 +84,8 @@ var results = do_something(3);
 
 Rust:
 
+Note again that the concept of "hoisting" doesn't really exist in Rust; all function definitions are scoped.
+
 ```rust
 // Function definition: takes an integer argument, returns an integer
 fn do_something(some_argument: i32) -> i32 {
```

Comment:
Ah, maybe make it clear that `: i32` is the type. Would be nice to maybe add a little table of the possible types for numbers somewhere? Is there a generic "number" type?